CMakeLists.txt

```diff
@@ -1018,21 +1018,22 @@ configure_file(
 if(CONFIG_CPU_HAS_MPU AND CONFIG_USERSPACE)
 
     if(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_APP_SHARED_MEM)
-        set(GEN_APP_SMEM $ENV{ZEPHYR_BASE}/scripts/gen_app_smem.py)
+        set(GEN_APP_SMEM $ENV{ZEPHYR_BASE}/scripts/gen_app_partitions.py)
         set(APP_SMEM_LD "${PROJECT_BINARY_DIR}/include/generated/app_smem.ld")
-        set(OBJ_FILE_DIR "${PROJECT_BINARY_DIR}/../")
 
         add_custom_target(
           ${APP_SMEM_DEP} ALL
-          DEPENDS zephyr_prebuilt
+	  DEPENDS app
+	  DEPENDS ${PRIV_STACK_DEP}
          )
 
         add_custom_command(
           TARGET ${APP_SMEM_DEP}
           COMMAND ${PYTHON_EXECUTABLE} ${GEN_APP_SMEM}
-          -d ${OBJ_FILE_DIR}
+          -d ${APPLICATION_SOURCE_DIR}
```

Comment:
APPLICATION_SOURCE_DIR is not (necessarily) the directory that contains the object files.